ddl/ddl_api.go

```diff
@@ -1096,6 +1096,68 @@ func checkConstraintNames(constraints []*ast.Constraint) error {
 	return nil
 }
 
+func setTableAutoRandomBits(tbInfo *model.TableInfo, colDefs []*ast.ColumnDef) error {
+	allowAutoRandom := config.GetGlobalConfig().Experimental.AllowAutoRandom
+	pkColName := tbInfo.GetPkName()
+	return forEachColWithAutoRandomOpt(colDefs, func(col *ast.ColumnDef, autoRandomBits uint64) error {
+		if !allowAutoRandom {
+			return ErrInvalidAutoRandom.GenWithStackByArgs(autoid.AutoRandomExperimentalDisabledErrMsg)
+		}
+
+		if tbInfo.PKIsHandle && col.Name.Name.L == pkColName.L {
```

Comment:
Maybe we can use `mysql.HasPriKeyFlag(col.Flag) ` instead of `col.Name.Name.L == pkColName.L`. Then we needn't to call `GetPkName`.

Reply:
`mysql.HasPriKeyFlag()` is not available in `ColDef`, because the information of primary key is not stored in `ColDef.Flag`.